extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -503,79 +523,33 @@ function processDropdownOptionsTypeField(context: FieldContext): azdata.DropDown
 		label: context.fieldInfo.label
 	});
 	dropdown.fireOnTextChange = true;
-	context.onNewInputComponentCreated(context.fieldInfo.variableName!, { component: dropdown });
+	context.onNewInputComponentCreated(context.fieldInfo.variableName || context.fieldInfo.label, { component: dropdown });
```

Comment:
variableName is not necessarily defined for all fields so where there is no variable name, use the field label as they 'key' for maintaining the fields collection.

Reply:
Hm, would almost rather see a required ID given to each field. But this is ok for now - something to consider for a follow up task.